Tests/Engine/InvokeScriptAnalyzer.tests.ps1

```diff
@@ -551,4 +551,30 @@ Describe "Test -EnableExit Switch" {
             "$result" | Should -Not -BeLike $reportSummaryFor1Warning 
         }
     }
+
+    if (!$testingLibraryUsage) {
+        Describe "Handles parse errors due to unknown types" {
+            $script = @'
+                using namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels
+                using namespace Microsoft.Azure.Commands.Common.Authentication.Abstractions
+                Import-Module "AzureRm"
+                class MyClass { [IStorageContext]$StorageContext } # This will result in a parser error due to [IStorageContext] type that comes from the using stetement but is not known at parse time
+                gci # Produce AvoidAlias rule
+'@
+            $assertionMessage = 'PSSA should analyze the whole script after the parse error on [IStorageContext] and find the PSAvoidUsingCmdletAliases warning due to gci'
+            It "does not throw and detect one expected warning after the parse error has occured when using -ScriptDefintion parameter set" {
+                $warnings = Invoke-ScriptAnalyzer -ScriptDefinition $script
+                $warnings.Count | Should -Be 1 -Because $assertionMessage
+                $warnings.RuleName | Should -Be 'PSAvoidUsingCmdletAliases' -Because $assertionMessage
```

Comment:
I know that there a bunch of these sorts of tests which check the message, but I'm not a fan. If we want to have tests for the message catalog, we can do that explicitly. Is there another way to check this? Checking a rulename is not subject to localization.

Reply:
This is just checking for the rule type/name, not the message itself, the `-Because` operator is for specifying additional details in the assertion message why the test was expected to fail if it fails, not an assertion.